### DISCUSSION TITLE
How to get images from server (in C)?

### DISCUSSION THREAD

**Discussion by luishhmnn** [microsoft/AirSim#4046] [Support Q&A] [answered]
I need the images of the AirSim server in my C program. I know that there is no C API. However, the server does send the images as a UDP message. Is it possible to receive these messages using C? Has anyone done this before?

**Comment by jonyMarino** [comment]
Hi @luishhmnn. C++ is not suitable for you? what are you trying to accomplish?

**Comment by luishhmnn** [comment, marked as answer]
Hi @jonyMarino. My problem is solved, but I can not delete the question.

I am using a framework written in C. The wish was that I also write the code for the communication with AirSim in C. I solved the problem by creating a C wrapper. I think this is ok.

**Reply by jonyMarino** [reply]
@luishhmnn Good to hear. How did you solve it? Have you added an extern "C" block for the AirSim client and used it inside your framework?
Just curious, what is the framework you are using?

**Reply by luishhmnn** [reply]
Exactly, I used an extern "C" block.
